### ISSUE TITLE
NetCDF tests failing

### ISSUE THREAD

**Issue by max-sixty** [pydata/xarray#3308]
(edit: original failure was mistaken)
Does anyone know off hand why [this](https://dev.azure.com/xarray/xarray/_build/results?buildId=798) is failing?

```
ResolvePackageNotFound: 
  - pandas=0.19
  - python=3.5.0
```

Worst case we could drop it... https://github.com/pydata/xarray/issues/3293

**Comment by crusaderky** [comment]
Looks like a very old fork, the min python version in master is 3.5.3. Try merging master in...

**Comment by max-sixty** [comment]
Thanks @crusaderky, I was confusing for another failure. This is the failure on master:

```
_____________ TestH5NetCDFData.test_complex[None-FutureWarning-1] ______________

self = <xarray.tests.test_backends.TestH5NetCDFData object at 0x7f39ad21e358>
invalid_netcdf = None, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
_____________ TestH5NetCDFData.test_complex[False-FutureWarning-1] _____________

self = <xarray.tests.test_backends.TestH5NetCDFData object at 0x7f39ae3fc668>
invalid_netcdf = False, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
__________________ TestH5NetCDFData.test_complex[True-None-0] __________________

self = <xarray.tests.test_backends.TestH5NetCDFData object at 0x7f39ad2058d0>
invalid_netcdf = True, warns = None, num_warns = 0

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 1 == 0
E        +  where 1 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
__________ TestH5NetCDFFileObject.test_complex[None-FutureWarning-1] ___________

self = <xarray.tests.test_backends.TestH5NetCDFFileObject object at 0x7f39ad7402e8>
invalid_netcdf = None, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
__________ TestH5NetCDFFileObject.test_complex[False-FutureWarning-1] __________

self = <xarray.tests.test_backends.TestH5NetCDFFileObject object at 0x7f39ad545940>
invalid_netcdf = False, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
_______________ TestH5NetCDFFileObject.test_complex[True-None-0] _______________

self = <xarray.tests.test_backends.TestH5NetCDFFileObject object at 0x7f39ae20ae80>
invalid_netcdf = True, warns = None, num_warns = 0

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 1 == 0
E        +  where 1 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
__________ TestH5NetCDFViaDaskData.test_complex[None-FutureWarning-1] __________

self = <xarray.tests.test_backends.TestH5NetCDFViaDaskData object at 0x7f39ad740b70>
invalid_netcdf = None, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
_________ TestH5NetCDFViaDaskData.test_complex[False-FutureWarning-1] __________

self = <xarray.tests.test_backends.TestH5NetCDFViaDaskData object at 0x7f39ae4a8d30>
invalid_netcdf = False, warns = <class 'FutureWarning'>, num_warns = 1

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 2 == 1
E        +  where 2 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
______________ TestH5NetCDFViaDaskData.test_complex[True-None-0] _______________

self = <xarray.tests.test_backends.TestH5NetCDFViaDaskData object at 0x7f39ad4ee3c8>
invalid_netcdf = True, warns = None, num_warns = 0

    @pytest.mark.filterwarnings("ignore:complex dtypes are supported by h5py")
    @pytest.mark.parametrize(
        "invalid_netcdf, warns, num_warns",
        [(None, FutureWarning, 1), (False, FutureWarning, 1), (True, None, 0)],
    )
    def test_complex(self, invalid_netcdf, warns, num_warns):
        expected = Dataset({"x": ("y", np.ones(5) + 1j * np.ones(5))})
        save_kwargs = {"invalid_netcdf": invalid_netcdf}
        with pytest.warns(warns) as record:
            with self.roundtrip(expected, save_kwargs=save_kwargs) as actual:
                assert_equal(expected, actual)
>       assert len(record) == num_warns
E       assert 1 == 0
E        +  where 1 = len(WarningsChecker(record=True))

xarray/tests/test_backends.py:2185: AssertionError
```

**Comment by dcherian** [comment]
This is #3300 which is fixed by #3301

**Comment by max-sixty** [comment]
Thanks @dcherian
